compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -286,19 +286,35 @@ pub trait Machine<'mir, 'tcx>: Sized {
     ) -> Pointer<Self::PointerTag>;
 
     /// "Int-to-pointer cast"
-    fn ptr_from_addr(
+    fn ptr_from_addr_cast(
         ecx: &InterpCx<'mir, 'tcx, Self>,
         addr: u64,
     ) -> Pointer<Option<Self::PointerTag>>;
```

Comment:
Hm, looks like we could have a default impl here that returns `None` provenance?

In fact, why do we even need this as a machine hook? Looks like *all* implementations are the same?

Reply:
Sorry I meant to attach this to `ptr_from_addr_transmute`.